packages/web/src/services/track-download.ts

```diff
@@ -10,29 +14,57 @@ function isMobileSafari() {
   )
 }
 
+function browserDownload({
+  url,
+  filename
+}: {
+  url: string
+  filename?: string
+}) {
+  if (document) {
+    const link = document.createElement('a')
+    link.href = url
+    // taget=_blank does not work on ios safari and will cause the download to be
+    // unresponsive.
+    if (!isMobileSafari()) {
+      link.target = '_blank'
+    }
+    link.download = filename ?? ''
+    link.click()
+    link.remove()
+  } else {
+    throw new Error('No document found')
+  }
+}
+
 class TrackDownload extends TrackDownloadBase {
-  async downloadTrack({ url, filename }: { url: string; filename: string }) {
-    const response = await window.fetch(url)
-    if (!response.ok) {
+  async downloadTrack({ files, rootDirectoryName }: DownloadTrackArgs) {
+    const responsePromises = files.map(
+      async ({ url }) => await window.fetch(url)
+    )
+    const responses = await Promise.all(responsePromises)
+    if (!responses.every((response) => response.ok)) {
       throw new Error('Download unsuccessful')
     }
-
-    const downloadURL = (url: string, filename: string) => {
-      if (document) {
-        const link = document.createElement('a')
-        link.href = url
-        // taget=_blank does not work on ios safari and will cause the download to be
-        // unresponsive.
-        if (!isMobileSafari()) {
-          link.target = '_blank'
-        }
-        link.download = filename
-        link.click()
-      } else {
-        throw new Error('No document found')
-      }
+    let url
+    if (files.length === 1) {
+      url = responses[0].url
+    } else {
+      if (!rootDirectoryName)
+        throw new Error(
+          'rootDirectory must be supplied when downloading multiple files'
+        )
+      const blob = await downloadZip(
```

Comment:
If I understand correctly, the way we're doing this is:
1. Fetch all files in parallel (into memory?)
2. Pass the finished responses to `downloadZip` and ask it to generate a blob
3. Use the download util to execute a download of said blob.

I'm a tad concerned about queueing an arbitrary number of (potentially quite large) files for download simultaneously, and then zipping them all together at the end. client-zip appears to be [designed as a _streaming_ generator](https://github.com/Touffy/client-zip?tab=readme-ov-file#usage) (see the section beginning "Unless your inputs list is quite small..."). The idea there being that we would provide some type of lazy generator that can be used to fetch the files in appropriate batches (such as with the [companion streaming library](https://github.com/Touffy/dl-stream) ). I _think_ the streaming case is probably more noticeable for large _numbers_ of files and maybe not just large files in general.  But it's worth considering:

* How large do we expect the data to be in the multi-file case? Have we tested this with slow network emulation in dev tools?
* What does our experience look like for downloads of 100+MB of content (possible or likely in the original stems case)? Do we have any way to provide progress feedback?

This doesn't necessarily have to be solved here, just wanted to put it out for consideration. Could be a follow-on after the feature is shipped.